### PR TITLE
Fix HDFSerializer.__getitem__

### LINKED ISSUE #811: Optimizer's states are serialized outside user-specified group
When user manually creates `HDF5Serializer` with user-specified group and saves an instance of `optimizer`, group hierarchy is created outside the group. See the code below for details.

```
In [2]: import chainer
In [3]: from chainer.link import Chain
In [4]: from chainer.links import Linear
In [5]: from chainer.optimizers import AdaDelta
In [6]: from chainer.serializers import HDF5Serializer
In [7]: import h5py
In [8]: chain = Chain(linear=Linear(100, 10))
In [9]: opt = AdaDelta()
In [10]: opt.setup(chain)
In [11]: h5 = h5py.File("test.h5")
In [12]: ls
test.h5
In [13]: h5.keys()
Out[13]: []
In [14]: group = h5.create_group("test")
In [15]: s = HDF5Serializer(group)
In [16]: s.save(opt)
In [17]: h5.keys()
Out[17]: [u'linear', u'test']  # 'linear' is supposed to create under 'test'
In [18]: h5['test'].keys()
Out[18]: [u'epoch', u't']
In [19]: h5['linear'].keys()
Out[19]: [u'W', u'b']
```

This is reported by Junya Norimatsu. Thank you!

### PR DESCRIPTION
**Pull request by delta2323**
Fix #811

### CODE REVIEW

File: tests/chainer_tests/serializers_tests/test_hdf5.py [hunk 1]
```diff
@@ -165,4 +168,59 @@ def test_load(self):
...
+        serializer = self._prepare_serializer()
+        serializer.save(self.parent)
+        self.assertSetEqual(set(self.h5.keys()), set(('test',)))
+        self._check_group(self.h5['test'], ('Wp',))
+
+    def test_save_optimizer(self):
+        serializer = self._prepare_serializer()
+        serializer.save(self.optimizer)
+        self.assertSetEqual(set(self.h5.keys()), set(('test',)))
+        self._check_group(self.h5['test'], ('Wp', 'epoch', 't'))
```
Comment (okuta) [inline tests/chainer_tests/serializers_tests/test_hdf5.py]: Please close `self.h5`.

File: tests/chainer_tests/serializers_tests/test_hdf5.py [hunk 2]
```diff
@@ -165,4 +168,59 @@ def test_load(self):
...
+        self.assertSetEqual(set(h5['child'].keys()),
+                            set(('linear', 'Wc')))
+        self.assertSetEqual(set(h5['child']['linear'].keys()),
+                            set(('W', 'b')))
+
+    def test_save_chain(self):
+        serializer = self._prepare_serializer()
+        serializer.save(self.parent)
+        self.assertSetEqual(set(self.h5.keys()), set(('test',)))
+        self._check_group(self.h5['test'], ('Wp',))
```
Comment (okuta) [inline tests/chainer_tests/serializers_tests/test_hdf5.py]: Please close `self.h5`.

**Comment by delta2323** [comment]
@okuta  I fixed the code.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**